### ISSUE TITLE
Rails 5/Devise 4 deprecations

### ISSUE THREAD

**Issue by connorshea** [tinfoil/devise-two-factor#68]
These occur when we run our test suite with Rails 5.0.0.beta2, Devise 4.0.0.rc1, and a fork of devise-two-factor that allows Devise 4 in the gemspec:

```
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from <top (required)> at /home/ubuntu/ImageHex/vendor/bundle/ruby/2.3.0/bundler/gems/devise-two-factor-0fa0a6914739/lib/devise_two_factor/models/two_factor_authenticatable.rb:1)
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from <top (required)> at /home/ubuntu/ImageHex/vendor/bundle/ruby/2.3.0/bundler/gems/devise-two-factor-0fa0a6914739/lib/devise_two_factor/models/two_factor_authenticatable.rb:1)
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from <top (required)> at /home/ubuntu/ImageHex/vendor/bundle/ruby/2.3.0/bundler/gems/devise-two-factor-0fa0a6914739/lib/devise_two_factor/models/two_factor_authenticatable.rb:1)
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from <top (required)> at /home/ubuntu/ImageHex/vendor/bundle/ruby/2.3.0/bundler/gems/devise-two-factor-0fa0a6914739/lib/devise_two_factor/models/two_factor_authenticatable.rb:1)
```

Command used is `bundle exec rspec --color --require spec_helper --require rails_helper --format documentation spec` on Circle CI. Can provide more information if needed.

**Comment by connorshea** [comment]
cc: @bsedat, don't mean to be intrusive or anything, just figured this was important.

**Comment by bsedat** [comment]
Good finds! No need for the extra CC, I get notified for repo activity automatically :smile:

I don't have any bandwidth at the moment to fix this, but happy to look at any pull requests. We'll need to bump the minimum ruby version to be 2.0.0+ in order to have access to `Module#prepend`

**Comment by connorshea** [comment]
Ah, wasn't sure :) I'll see about opening a PR to make the gem compatible with Devise 4.0, but I'm not much of a Ruby expert so I probably won't be able to fix the deprecation warnings.

**Comment by connorshea** [comment]
Looks like this is a problem in attr_encrypted [that was just fixed in 2.0.0](https://github.com/attr-encrypted/attr_encrypted/blob/master/CHANGELOG.md). I'll test devise-two-factor with the 2.0.0 version of the gem.

**Comment by connorshea** [comment]
It wasn't fixed in 2.0.0, I was mistaken. I've [opened an issue](https://github.com/attr-encrypted/attr_encrypted/issues/193) in the repository to get the deprecation warnings fixed. We should keep this issue open to update the gem to 2.x when it's fixed.

**Comment by connorshea** [comment]
I think we can close this, since it's not an issue with devise-two-factor itself.
